scrapy/code/groceries/groceries/spiders/safeway/scraper.py

```diff
@@ -0,0 +1,267 @@
+#
+        #check if it has a load-more button and then increment page number on it
+        if response.css('.primary-btn.btn.btn-default.btn-secondary.bloom-load-button').get() is not None:
+            path = response.css('[aria-current]:not(.menu-nav__sub-item) ::text').getall()
+            #print(f"path - {path} for url - {response.url}")
+            section = path[1]
+            section = section.strip()
+            subsection = path[-2]
+            subsection = subsection.strip()
+            category = lookup_category("",section,subsection)
+            next_page_url=get_next_pagination(self.page_str,response.url)
+            print (f'load-more-button. storing - {next_page_url}, section - {section}, subsection - {subsection}, category - {category}')
+            store_url(self.conn,next_page_url,self.store_id,category,section,subsection)
+
+    def parse(self, response):
+        page_1_str=self.page_str+"1"
+        this_url = trim_url(response.url,page_1_str)
+        print (f"inside parse for {this_url}")
+        self.scrape_urls(response)
+
+        # Only scrape pages that have the page_str in the url.
+        if this_url.find(self.page_str) != -1:
+            print (f"scraping for {this_url}")
+            items = response.css('product-item-v2')
+            print(f"length of items - {len(items)}")
+            metadata=get_url_metadata(self.cursor,this_url)
+            section=metadata[1]
+            subsection=metadata[2]
+            for item in items:
+                name = item.css('.product-title ::text').get()
+                price_strings = item.css('.product-price ::text').getall()
+                price = clean_string(price_strings[-1],['$'])
+                ppu = item.css('.product-price-qty ::text').get()
+                unit = self.collect_units(name)
+                #inspect_response(response,self)
+
+                if unit == "OZ" or unit == "LB":
+                    ounces = self.collect_ounces(name)
+                else:
+                    ounces = 0
+                print (f"yielding - {name}, {price}, {ppu}, {ounces}, {unit}")
+                yield{
+                  "name": name,
+                  "price": price,
+                  "ounces": ounces,
+                  "unit": unit,
+                  "price-per-unit": ppu,
+                  "url": this_url,
+                  "section": section,
+                  "subsection": subsection
+                }
+
+        #Basically the website redirects us to the url and page_1_str, which isn't added to our database
+        # So we trim that off so we can get the url in our database
+        finish_url(self.conn,self.store_id,this_url)
+        print("finishing url - " + this_url)
+        next_url = get_next_url(self.cursor, 1)
+        if next_url is None:
+            print ("Next url is none therefore we must be finished ! ")
+            return
+        #elif next_url.find(self.page_str) != -1:
```

Comment:
Do we need this?

Reply:
Yeah - if get_next_url returns None - then its finished (there are no more urls to scrape) else it keeps scraping

Reply:
I'll remove the comment though